src/test/get_pipeline_metadata_response_test.cpp

```diff
@@ -0,0 +1,264 @@
+    ASSERT_EQ(ovms::GetModelMetadataImpl::buildResponse(pipelineDefinition, &response, manager), ovms::StatusCode::OK);
+
+    tensorflow::serving::SignatureDefMap def;
+    response.metadata().at("signature_def").UnpackTo(&def);
+
+    EXPECT_EQ(def.signature_def_size(), 1);
+}
+
+TEST_F(GetPipelineMetadataResponse, HasCorrectSignatureDefName) {
+    ASSERT_EQ(ovms::GetModelMetadataImpl::buildResponse(pipelineDefinition, &response, manager), ovms::StatusCode::OK);
+
+    tensorflow::serving::SignatureDefMap def;
+    response.metadata().at("signature_def").UnpackTo(&def);
+
+    EXPECT_NE(
+        def.signature_def().find("serving_default"),
+        def.signature_def().end());
+}
+
+TEST_F(GetPipelineMetadataResponse, HasCorrectTensorNames) {
+    ASSERT_EQ(ovms::GetModelMetadataImpl::buildResponse(pipelineDefinition, &response, manager), ovms::StatusCode::OK);
+
+    tensorflow::serving::SignatureDefMap def;
+    response.metadata().at("signature_def").UnpackTo(&def);
+
+    const auto& inputs = ((*def.mutable_signature_def())["serving_default"]).inputs();
+    const auto& outputs = ((*def.mutable_signature_def())["serving_default"]).outputs();
+
+    EXPECT_EQ(inputs.size(), 3);
+    EXPECT_EQ(outputs.size(), 3);
+
+    EXPECT_EQ(
+        inputs.at("Input_FP32_1_3_224_224").name(),
+        "Input_FP32_1_3_224_224");
+    EXPECT_EQ(
+        inputs.at("Input_U8_1_3_62_62").name(),
+        "Input_U8_1_3_62_62");
+    EXPECT_EQ(
+        inputs.at("Input_Unspecified").name(),
+        "Input_Unspecified");
+    EXPECT_EQ(
+        outputs.at("Output_I32_1_2000").name(),
+        "Output_I32_1_2000");
+    EXPECT_EQ(
+        outputs.at("Output_FP32_2_20_3").name(),
+        "Output_FP32_2_20_3");
+    EXPECT_EQ(
+        outputs.at("Output_Unspecified").name(),
+        "Output_Unspecified");
+}
+
+TEST_F(GetPipelineMetadataResponse, HasCorrectPrecision) {
+    ASSERT_EQ(ovms::GetModelMetadataImpl::buildResponse(pipelineDefinition, &response, manager), ovms::StatusCode::OK);
+
+    tensorflow::serving::SignatureDefMap def;
+    response.metadata().at("signature_def").UnpackTo(&def);
+
+    const auto& inputs = ((*def.mutable_signature_def())["serving_default"]).inputs();
+    const auto& outputs = ((*def.mutable_signature_def())["serving_default"]).outputs();
+
+    EXPECT_EQ(
+        inputs.at("Input_FP32_1_3_224_224").dtype(),
+        tensorflow::DT_FLOAT);
+    EXPECT_EQ(
+        inputs.at("Input_U8_1_3_62_62").dtype(),
+        tensorflow::DT_UINT8);
+    EXPECT_EQ(
+        inputs.at("Input_Unspecified").dtype(),
+        tensorflow::DT_INVALID);
+    EXPECT_EQ(
+        outputs.at("Output_I32_1_2000").dtype(),
+        tensorflow::DT_INT32);
+    EXPECT_EQ(
+        outputs.at("Output_FP32_2_20_3").dtype(),
+        tensorflow::DT_FLOAT);
+    EXPECT_EQ(
+        outputs.at("Output_Unspecified").dtype(),
+        tensorflow::DT_INVALID);
+}
+
+TEST_F(GetPipelineMetadataResponse, HasCorrectShape) {
```

Comment:
L77-190 tests could be all in one fixture - all tests perform exactly the same at the beginning:
ASSERT_EQ(ovms::GetModelMetadataImpl::buildResponse(pipelineDefinition, &response, manager), ovms::StatusCode::OK);
Then just check different things. No need to call this function so many times with the same inputs.

You could move this to setUp(), and all the other tests to other fixture

Reply:
Done, both for pipelines and models (get_model_metadata_response_test.cpp aswell)